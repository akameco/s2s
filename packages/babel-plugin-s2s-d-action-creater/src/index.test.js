// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const filename = path.resolve(process.cwd(), 'app', 'counter', 'actionTypes.js')

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: { filename },
  tests: [
    {
      title: 'one function',
      code: `
// @flow
import * as constants from './constants'

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;
      `,
    },
    {
      title: 'some function',
      code: `
import * as constants from './constants'

export type Increment = {
  type: typeof constants.INCREMENT,
}

export type Decrement = {
  type: typeof constants.DECREMENT,
}

export type Action = Increment | Decrement
      `,
    },
    {
      title: 'has prop',
      code: `
// @flow
import * as constants from './constants'

export type Add = {
  type: typeof ADD,
  payload: number,
}

export type Action = Add
    `,
    },
    {
      title: 'another import',
      code: `
// @flow
import type { Count } from './types'

export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  n: number,
  payload: Count,
}

export type Action = Add
    `,
    },
  ],
})
