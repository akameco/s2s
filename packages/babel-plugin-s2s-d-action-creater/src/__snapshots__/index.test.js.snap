// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default another import: another import 1`] = `
"
// @flow
import type { Count } from './types'

export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  n: number,
  payload: Count,
}

export type Action = Add

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Action } from './types';
import type { Count } from './types';

export const add = (n: number, payload: Count): Action => {
  return {
    type: constants.ADD,
    n,
    payload
  };
};
"
`;

exports[`default has prop: has prop 1`] = `
"
// @flow
import * as constants from './constants'

export type Add = {
  type: typeof ADD,
  payload: number,
}

export type Action = Add

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Action } from './types';
import * as constants from './constants';

export const add = (payload: number): Action => {
  return {
    type: constants.ADD,
    payload
  };
};
"
`;

exports[`default one function: one function 1`] = `
"
// @flow
import * as constants from './constants'

export type Increment = {
  type: typeof constants.INCREMENT
};

export type Action = Increment;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Action } from './types';
import * as constants from './constants';

export const increment = (): Action => {
  return {
    type: constants.INCREMENT
  };
};
"
`;

exports[`default some function: some function 1`] = `
"
import * as constants from './constants'

export type Increment = {
  type: typeof constants.INCREMENT,
}

export type Decrement = {
  type: typeof constants.DECREMENT,
}

export type Action = Increment | Decrement

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Action } from './types';
import * as constants from './constants';

export const increment = (): Action => {
  return {
    type: constants.INCREMENT
  };
};

export const decrement = (): Action => {
  return {
    type: constants.DECREMENT
  };
};
"
`;
